### DISCUSSION TITLE
Welcome to Python Type School!

### DISCUSSION THREAD

**Discussion by srittau** [python/typing#826] [Announcements]
## Welcome to Python Type School!

This is a place where you can ask questions about optional typing with Python. 🐍 Any questions are welcome, but you might want to look through existing discussions, your question might already have an answer! We also have a [gitter channel](https://gitter.im/python/typing) if you are looking for a more interactive form of help.

Our discussion categories are:

* [Q&A](/python/typing/discussions/categories/q-a) where you can ask question about typing or help others.
* [Reviews](/python/typing/discussions/categories/reviews) where you can ask others to review your typing-related patches or pull requests.
* [General](/python/typing/discussions/categories/general) for all other typing-related discussions. But please note that ideas about improving Python's typing are best directed to the [typing-sig mailing list](https://mail.python.org/archives/list/typing-sig@python.org/). Historically, [the issues](/python/typing/issues) in this repository had discussions about new features as well, but new discussions should be started on the mailing list.

For more information about typing see the following resources:

* [Documentation for the `typing` module](https://docs.python.org/3/library/typing.html)
* [The mypy documentation](https://mypy.readthedocs.io/en/stable/)

**Comment by juanchoflorez** [comment]
Is there a python api for mypy that can be used to get types of variables?
Context:
I want to infer the type of variables inside a python script, for example, I would like to be able to know that `simple_obj` is of type `ObjWithMethod` in the following script:
```
class ObjWithTask:
    def __init__(self, ii: int = 0):
        self._ii = ii

    def get_id(
        self,
    ):
        return self._ii

def my_func():
    simple_obj = ObjWithTask(ii=3)
    object_id = simple_obj.get_id()
    return [object_id]

```
Is it possible to do? I know that VSCode is able to infer that, however I am not sure how it does it, if it uses proprietary code from Pylance or Pyright. Any comments or suggestions are welcome

**Reply by jeremyn** [reply]
@juanchoflorez This is just an announcement thread for this board, it's not really a place for questions. Could you repost your comment in a new thread? Feel free to @ me there, I have an idea that might help you.

**Comment by jeremyn** [comment]
@srittau Please consider [pinning this discussion](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#pinning-a-discussion) to the top of the list.

**Reply by srittau** [reply]
The discussion is actually pinned. Unfortunately, pinned discussions are displayed in form of a large header, so that it's easy to overlook, despite them being so prominent.

**Reply by jeremyn** [reply]
Hah, I see, I had no idea that header was even clickable. Good old [banner blindness](https://en.wikipedia.org/wiki/Banner_blindness).

**Reply by srittau** [reply]
I have to admit, when I just checked, I didn't see the pinned discussion either, only when I noticed that it said "Unpin discussion" instead of "Pin discussion" I looked closer. I prefer the typical pinning of other forums, where the pinned discussions are just at the front of the list, maybe with a slightly different background.
